### ISSUE TITLE
Parameter to shorten comparison lines

### ISSUE THREAD

**Issue by romanhaa** [kassambara/ggpubr#285]
Hi!

Is there a way to slightly shorten the comparison lines so that one can put multiple comparisons on the same height without having overlaps of the lines?

This is what I have in mind (and I was able to do it with manual annotations through the `ggsignif` package):

![Screenshot 2020-05-27 at 12 51 57](https://user-images.githubusercontent.com/18103560/83010717-fcd99a80-a018-11ea-910d-15364c52836e.png)

And this is what I was able to do with `stat_compare_means()`,

![Screenshot 2020-05-27 at 12 51 40](https://user-images.githubusercontent.com/18103560/83010859-33171a00-a019-11ea-8ce2-58b1def7e310.png)

It would be great if there was a parameter to shorten the comparison lines so that manually preparing a data frame with all the annotations and x positions from and to where the lines should be drawn. I can save some space above the box plot and it looks prettier too, at least in my opinion. Actually, I first thought `bracket.width` was what I was looking for but then realized that it was for the line width :/

Cheers,
Roman

**Comment by kassambara** [comment]
With the latest dev version of the ggpubr and rstatix package, you can do this:

``` r
suppressPackageStartupMessages(library(ggpubr))
suppressPackageStartupMessages(library(rstatix))
# Data preparation
df <- ToothGrowth
df$dose <- factor(df$dose)
# Statistical test 
stat.test <- df %>%
  t_test(len ~ dose)
stat.test
#> # A tibble: 3 x 10
#>   .y.   group1 group2    n1    n2 statistic    df        p    p.adj p.adj.signif
#> * <chr> <chr>  <chr>  <int> <int>     <dbl> <dbl>    <dbl>    <dbl> <chr>       
#> 1 len   0.5    1         20    20     -6.48  38.0 1.27e- 7 2.54e- 7 ****        
#> 2 len   0.5    2         20    20    -11.8   36.9 4.40e-14 1.32e-13 ****        
#> 3 len   1      2         20    20     -4.90  37.1 1.91e- 5 1.91e- 5 ****

# Autocompute p-value position and add onto the box-plot
stat.test <- stat.test %>% add_xy_position(x = "dose")
ggboxplot(df, x = "dose", y = "len")+ 
  stat_pvalue_manual(stat.test, tip.length = 0.01)
```

![](https://i.imgur.com/fT54ioR.png)

``` r

# Specify manually p-values y position
# Shorten brackets
ggboxplot(df, x = "dose", y = "len")+ 
  stat_pvalue_manual(
    stat.test, tip.length = 0.01,
    y.position = c(35, 38, 35), bracket.shorten = 0.1
    )
```

![](https://i.imgur.com/iAwHZGb.png)

Thank you for the suggestion

**Comment by romanhaa** [comment]
Thanks @kassambara for the feedback. So I guess it was a little useless that I implemented it in ggsignif as well. I was going to create a pull request for the ggsignif package (where I added an `extend_line` parameter that can be used to extend and shorten the bracket line) and then also one for ggpubr to make it compatible with the change. I'm just a little busy at the moment so I was going to wait. But good to see that you already did this.
